octue/resources/dataset.py

```diff
@@ -35,12 +36,32 @@ def __init__(self, id=None, logger=None, path=None, path_from=None, base_from=No
             else:
                 self.files.append(Datafile(**fi, path_from=self, base_from=self))
 
+        Filterable.__init__(self, filters=self._build_filters())
         self.__dict__.update(**kwargs)
 
     @property
     def name(self):
         return str(os.path.split(self.path)[-1])
 
+    def __iter__(self):
+        yield from self.files
+
+    def __len__(self):
+        return len(self.files)
+
+    def _build_filters(self):
+        return {
+            "name__icontains": ("files", lambda file, filter_value: filter_value.lower() in file.name.lower()),
+            "name__contains": ("files", lambda file, filter_value: filter_value in file.name),
+            "name__ends_with": ("files", lambda file, filter_value: file.name.endswith(filter_value)),
+            "name__starts_with": ("files", lambda file, filter_value: file.name.startswith(filter_value)),
+            "tag__exact": ("files", lambda file, filter_value: filter_value in file.tags),
+            "tag__starts_with": ("files", lambda file, filter_value: file.tags.starts_with(filter_value)),
+            "tag__ends_with": ("files", lambda file, filter_value: file.tags.ends_with(filter_value)),
+            "tag__contains": ("files", lambda file, filter_value: file.tags.contains(filter_value)),
+            "sequence__notnone": ("files", lambda file, filter_value: file.sequence is not None),
```

Comment:
I think it would be quite easy to define standard set of these if we narrow the scope to iterables and relatively simple filtering criteria. The reason I didn't pursue a single set of filters in the end is that I thought they would use cases that can be easily constrained into them. However, I did still keep a standard set of "base filters" - take a look at the top of the `Filterable` class